pkg/address/validate_address.go

```diff
@@ -0,0 +1,59 @@
+package address
+
+import (
+	"fmt"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+)
+
+// ValidateAddressForChain validates the address for the chain
+// NOTE: since these checks are currently not used, we don't provide additional chains for simplicity
+// https://github.com/zeta-chain/node/issues/2234
+// https://github.com/zeta-chain/node/issues/2385
+// NOTE: We should eventually not using these hard-coded checks at all for same reasons as above
+
+// TODO : Use this function to validate Sender and Receiver address for CCTX
+// https://github.com/zeta-chain/node/issues/2697
+func ValidateAddressForChain(address string, chainID int64, additionalChains []chains.Chain) error {
+	chain, found := chains.GetChainFromChainID(chainID, additionalChains)
+	if !found {
+		return fmt.Errorf("chain id %d not supported", chainID)
+	}
+	switch chain.Network {
+	case chains.Network_eth:
+		return ValidateEthereumAddress(address)
+	case chains.Network_zeta:
+		return nil
```

Comment:
This should also validate for evm

Reply:
This function is not used right now
It would depend on the v2 protocol refactor on how this value is utilised .
Presently for btc inbound ,the receiver is set as `FromAddress` which would cause this check to fail and disable btc deposits .
Overall we should test this via end to end tests when using this function